apps/04TowersOfHanoi.js

```diff
@@ -16,23 +26,85 @@ function printStacks() {
     console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+
+
+// Instructions for Spec 1
+// below is our function declaration, or function statement
+// function statements should contain:
+//
+//
+// 1. the name of the function  (we already have this)
+// 2. arguments in parentheses seperated by commas (we are given these, startStack and endStack)
+// 3. javascript statements that define the function enclosed in curly brackets(we are told to startStack.push() and endStack.pull())
+//
+//
+// After we define our function we then need to call this function from another function towersOfHanoi()
+// End Spec 1
+
+
+function movePiece(startStack, endStack) {
     // Your code here
 
+    // startStack.pop();
+    // stacks[startStack].pop()
+    var stackPiece = stacks[startStack].pop();
+    stacks[endStack].push(stackPiece);
+
 }
 
-function isLegal() {
-    // Your code here
 
+// Instructions for Spec 2
+// 1. the function isLegal() takes two arguments, startStack and endStack
+// 2. It should check to see if the block(stackPiece) being moved,
+// from startStack is smaller (less than operator) than the last the block in endStack
+// 3. return true if it is allowed
+// 4. return false if it is not
+// 5. if endStack has no block any block can be moved there (if endStack is empty the move is legal)
+//
+//
+//
+
+// what is a legal move?
+// if something exists in start stack and nothing exits in endStack then the
+
+function isLegal(startStack, endStack) {
+    // Your code here
+    var startStackArray = stacks[startStack];
+    var endStackArray   = stacks[endStack];
+
+    if (startStackArray.length === 0) {
+      return false;
+    }
+    else if (endStackArray.length === 0) {
+      return true;
+    }
+    else {
+      return startStackArray[startStackArray.length - 1] < endStackArray[endStackArray.length - 1];
```

Comment:
Its also helpful to draw on paper what this means. so if I follow this statements logic with my current stacks lookin like this:
```
a: [4,3,2]
b: [1]
c: []
```
im gonna try to move 2(a) onto 1(b)

startStackArray.length-1 will give me the 'top' of my stack, or '2', and endStackArray.length-1 give me the top of my other stack, or '1'...so using the statement:

2 < 1...This is False, and an illegal move. Which makes sense, cuz we cant move a bigger disk onto a smaller one.

If I was looking at this project with fresh eyes and limited experience in JavaScript syntax, I wouldn't have even thought of doing okay startStackArray.length-1...nope. What I would think is, how can I represent, in code, that this scenario: larger disk cannot be placed on smaller disk : would return a false? How can I represent with code, the 'top' of a stack, or the last element in the array?

Because the user has to follow the rules of just moving the top disks on a stack, I know I gotta check those to determine whether it will be legal or not. Here is how I would outline my isLegal function in pseudocode, then I would go back and try to do some real code by asking for syntax help:

```
function isLegal(startStack, endStack) {

//gotta find a way to relate startStack and endStack to the variable stacks in final result

if (startStack is empty){
    return illegal;
}
else if( endStack is empty) {
return legal;
}
else if(top of startStack < top of endstack){
return legal;
}
}
```

after writing that, i would try to get the correct syntax.